src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.BaseNameProvider.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FindSymbols
+{
+    internal partial class SymbolTreeInfo
+    {
+        /// <summary>
+        /// Used to produce the simple-full-name components of a type from metadata.
+        /// The name is 'simple' in that it does not contain things like backticks,
+        /// generic arguments, or nested type + separators.  Instead just hte name
+        /// of the type, any containing types, and the component parts of its namespace
+        /// are added.  For example, for the type "X.Y.O`1.I`2, we will produce [X, Y, O, I]
+        /// 
+        /// </summary>
+        private class BaseNameProvider : ISignatureTypeProvider<object>
```

Comment:
>object [](start = 64, length = 6)

This should be singleton. Use List<string> instead of object as the parameter and pass the list thru.